src/hal/halmodule.cc

```diff
@@ -74,15 +79,57 @@ bool from_python(PyObject *o, double *d) {
     return true;
 }
 
-bool from_python(PyObject *o, uint32_t *u) {
+bool from_python(PyObject *o, hal_u32_t *u) {
+    PyObject *tmp = 0;
+    long long l;
+    tmp = PyLong_Check(o) ? o : PyNumber_Long(o);
+    if(!tmp) goto fail;
+
+    l = PyLong_AsLongLong(tmp);
+    if(PyErr_Occurred()) goto fail;
+    if(l < 0 || l != (hal_u32_t)l) {
+        PyErr_Format(PyExc_OverflowError, "Value %lld out of range", l);
+        goto fail;
+    }
+
+    *u = l;
+    if(tmp != o) Py_XDECREF(tmp);
+    return true;
+fail:
+    if(tmp != o) Py_XDECREF(tmp);
+    return false;
+}
+
+bool from_python(PyObject *o, hal_s32_t *i) {
     PyObject *tmp = 0;
     long long l;
     tmp = PyLong_Check(o) ? o : PyNumber_Long(o);
     if(!tmp) goto fail;
 
     l = PyLong_AsLongLong(tmp);
     if(PyErr_Occurred()) goto fail;
-    if(l < 0 || l != (uint32_t)l) {
+    if(l != (hal_s32_t)l) {
```

Comment:
Should this not be hal_u32_t ?

Reply:
It should not :) On https://docs.python.org/3/c-api/long.html, it is long long PyLong_AsLongLong([PyObject](https://docs.python.org/3/c-api/structures.html#c.PyObject) *obj), so there is no particular need to cast this down, right? Anyway ...